init.lua

```diff
@@ -182,100 +207,208 @@ local cantLootID = 0
+                    data[param] = value;
+                end
+            elseif section == sec and param ~= nil then
+                data[param] = value;
+            end
+        end
+    end
+    file:close();
+    return data;
+end
+
+local function File_Exists(name)
+    local f = io.open(name, "r")
+    if f ~= nil then
+        io.close(f)
+        return true
+    else
+        return false
+    end
+end
 
-local function writeSettings()
-    for option,value in pairs(loot) do
+function loot.writeSettings()
+    for option, value in pairs(loot.Settings) do
         local valueType = type(value)
         if saveOptionTypes[valueType] then
-            mq.cmdf('/ini "%s" "%s" "%s" "%s"', loot.SettingsFile, 'Settings', option, value)
+            mq.cmdf('/ini "%s" "%s" "%s" "%s"', SettingsFile, 'Settings', option, value)
+            loot.Settings[option] = value
         end
     end
-    for option,value in pairs(BuyItems) do
+    for option, value in pairs(loot.BuyItems) do
         local valueType = type(value)
         if saveOptionTypes[valueType] then
-            mq.cmdf('/ini "%s" "%s" "%s" "%s"', loot.SettingsFile, 'BuyItems', option, value)
+            mq.cmdf('/ini "%s" "%s" "%s" "%s"', SettingsFile, 'BuyItems', option, value)
+            loot.BuyItems[option] = value
+        end
+    end
+    for option, value in pairs(loot.GlobalItems) do
+        local valueType = type(value)
+        if saveOptionTypes[valueType] then
+            mq.cmdf('/ini "%s" "%s" "%s" "%s"', LootFile, 'GlobalItems', option, value)
+            loot.modifyItem(option, value, 'Global_Rules')
+            loot.GlobalItems[option] = value
         end
     end
-
 end
 
-local function split(input, sep)
+function loot.split(input, sep)
     if sep == nil then
         sep = "|"
     end
-    local t={}
-    for str in string.gmatch(input, "([^"..sep.."]+)") do
+    local t = {}
+    for str in string.gmatch(input, "([^" .. sep .. "]+)") do
         table.insert(t, str)
     end
     return t
 end
 
-local function loadSettings()
-    local iniSettings = mq.TLO.Ini.File(loot.SettingsFile).Section('Settings')
-    local keyCount = iniSettings.Key.Count()
-    for i=1,keyCount do
-        local key = iniSettings.Key.KeyAtIndex(i)()
-        local value = iniSettings.Key(key).Value()
-        if key == 'Version' then
-            loot[key] = tostring(value)
-        elseif value == 'true' or value == 'false' then
-            loot[key] = value == 'true' and true or false
-        elseif tonumber(value) then
-            loot[key] = tonumber(value)
-        else
-            loot[key] = value
-        end
+function loot.UpdateDB()
+    loot.NormalItems = loot.load(LootFile, 'items')
+    loot.GlobalItems = loot.load(LootFile, 'GlobalItems')
+
+    local db = SQLite3.open(ItemsDB)
+    for k, v in pairs(loot.NormalItems) do
+        local stmt, err = db:prepare("INSERT INTO Normal_Rules (item_name, item_rule) VALUES (?, ?)")
```

Comment:
not sure how common a large loot file import like mine would be but i think this would be better inserting in batches vs. being frozen for several minutes of individual inserts.

Reply:
i have an idea to test where do the import in transaction batches up to 500 entries at once instead of per line.